ebpf/tracer.go

```diff
@@ -81,11 +93,79 @@ func NewTracer(config *Config) (*Tracer, error) {
 		return nil, fmt.Errorf("failed to init module: error guessing offsets: %v", err)
 	}
 
-	return &Tracer{m: m, config: config}, nil
+	tr := &Tracer{
+		m:           m,
+		config:      config,
+		closedConns: []ConnectionStats{},
+	}
+
+	tr.perfMap, err = tr.initPerfPolling()
+	if err != nil {
+		return nil, fmt.Errorf("could not start polling bpf events: %s", err)
+	}
+
+	return tr, nil
+}
+
+// initPerfPolling starts the listening on perf buffer events to grab closed connections
+func (t *Tracer) initPerfPolling() (*bpflib.PerfMap, error) {
+	closedChannel := make(chan []byte, 100)
+	lostChannel := make(chan uint64, 10)
+
+	pm, err := bpflib.InitPerfMap(t.m, string(tcpCloseEventMap), closedChannel, lostChannel)
+	if err != nil {
+		return nil, fmt.Errorf("error initializing perf map: %s", err)
+	}
+
+	pm.PollStart()
+
+	go func() {
+		// Stats about how much connections have been closed / lost
+		var closedCount, lostCount uint64
+		ticker := time.NewTicker(5 * time.Minute)
+
+		for {
+			select {
+			case c, ok := <-closedChannel:
+				if !ok {
+					return
+				}
+				closedCount++
+				stats := decodeRawTCPConn(c)
+
+				t.closedConnsLock.Lock()
+				t.closedConns = append(t.closedConns, stats)
+				t.closedConnsLock.Unlock()
+
+			case c, ok := <-lostChannel:
+				if !ok {
+					return
+				}
+				lostCount += c
+
+			case <-ticker.C:
+				log.Infof("Connection stats: %d lost, %d closed", lostCount, closedCount)
```

Comment:
Will this be too noisy / not useful to the user at the INFO level? Should be it DEBUG instead?